pkg/filter/range.go

```diff
@@ -0,0 +1,33 @@
+package filter
+
+import "github.com/mgtv-tech/redis-GunYu/pkg/redis"
+
+type Range struct {
+	Left, Right uint16
+}
+
+type RangeList struct {
+	list []*Range
+}
+
+func NewRangeList() *RangeList {
+	return &RangeList{
+		list: make([]*Range, 0),
+	}
+}
+
+func (rl *RangeList) IsSlotInList(key string) bool {
+	keySlot := redis.KeyToSlot(key)
```

Comment:
It would be best to optimize it:
- if the length of list is below a value(maybe 10 or 5) : it is ok to iterate the list
- if not : binary search is better